eng/scripts/Language-Settings.ps1

```diff
@@ -355,59 +355,99 @@ function PackageDependenciesResolve($artifactNamePrefix, $packageDirectory) {
+    }
+
+    $results += $isValid
+  }
+
+  $allValid = $results.Where({ $_ -eq $false }).Count -eq 0
+
+  return $allValid
 }
 
-function DockerValidation($packageName, $packageVersion, $groupId, $DocValidationImageId, $workingdirectory) 
-{
-  $output = docker run -v "${workingDirectory}:/workdir/out" `
-    -e TARGET_PACKAGE=$packageName -e TARGET_VERSION=$packageVersion -e TARGET_GROUP_ID=$groupId -t $DocValidationImageId 2>&1 
-  # The docker exit codes: https://docs.docker.com/engine/reference/run/#exit-status
-  # If the docker failed because of docker itself instead of the application, 
-  # we should skip the validation and keep the packages. 
-  $artifactNamePrefix = "${groupId}:${packageName}:${packageVersion}"
-  if ($LASTEXITCODE -eq 125 -Or $LASTEXITCODE -eq 126 -Or $LASTEXITCODE -eq 127) 
-  { 
-    LogWarning "The `docker` command does not work with exit code $LASTEXITCODE. Fall back to mvn install $artifactNamePrefix directly."
-    $output | Write-Host
-    FallbackValidation -artifactNamePrefix "$artifactNamePrefix" -workingDirectory $workingdirectory
-  }
-  elseif ($LASTEXITCODE -ne 0) 
-  { 
-    LogWarning "Package $artifactNamePrefix ref docs validation failed."
-    $output | Write-Host
+function DockerValidation ($packageInfos, $DocValidationImageId, $workingDirectory) {
+  Write-Host "Validating $($packageInfos.Length) package(s) using $DocValidationImageId."
+
+  $containerWorkingDirectory = '/workdir/out'
+  $configurationFileName = 'configuration.json'
+
+  $hostConfigurationPath = Join-Path $workingDirectory $configurationFileName
+
+  # Cannot use Join-Path because the container and host path separators may differ
+  $containerConfigurationPath = "$containerWorkingDirectory/$configurationFileName"
+  
+  $configuration = [ordered]@{
+    "output_path" = "docs-ref-autogen";
+    "packages" = @($packageInfos | ForEach-Object { [ordered]@{ 
+        packageGroupId = $_.Group;
+        packageArtifactId = $_.Name;
+        packageVersion = $_.Version;
+        packageDownloadUrl = 'https://repo1.maven.org/maven2'
+      } });
+  }
+
+  Set-Content -Path $hostConfigurationPath -Value ($configuration | ConvertTo-Json) | Out-Null
+
+  docker run -v "${workingDirectory}:${containerWorkingDirectory}" `
+    -e TARGET_CONFIGURATION_PATH=$containerConfigurationPath -t $DocValidationImageId 2>&1 `
+    | Where-Object { -not ($_ -match '^Progress .*B\s*$') } ` # Remove progress messages 
+    | Out-Host
+  
+  if ($LASTEXITCODE -ne 0) { 
+    LogWarning "The `docker` command failed with exit code $LASTEXITCODE."
+     
+    # The docker exit codes: https://docs.docker.com/engine/reference/run/#exit-status
+    # If the docker validation failed because of docker itself instead of the application, or if we don't know which
+    # package failed, fall back to mvn validation
+    if ($LASTEXITCODE -eq 125 -Or $LASTEXITCODE -eq 126 -Or $LASTEXITCODE -eq 127 -Or $packageInfos.Length -gt 1) { 
```

Comment:
Minor nitpick: Couldn't we just do **$LASTEXITCODE -In 125...127** instead of checking equality against 3 consecutive numbers?

Reply:
Could do.   I was trying to keep it as close to the original as possible:

``` powershell
if ($LASTEXITCODE -eq 125 -Or $LASTEXITCODE -eq 126 -Or $LASTEXITCODE -eq 127)
```

Reply:
fixed